### PR TITLE
Saving and loading setting presets

### PR DESCRIPTION
**Pull request by ChielBruin**
- Added setting loading and saving in the settings manager
- Added buttons in the interface that invoke this fnctionality

### CODE REVIEW

File: catkin_ws/src/kpr_interface/cfg/settings.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"maxCurvature": "100", "maxWeight": "4400", "minWeight": "300", "armSpeed": "100"}
```
Comment (MaxGroenenboom) [inline catkin_ws/src/kpr_interface/cfg/settings.json:1]: Why do you save these as strings? JSON does support numbers

File: catkin_ws/src/kpr_interface/srv/SettingsIO.srv [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Filepath to store the file to or load it from
+string filePath
+---
+# A status code that indicats the succes of the operation
```
Comment (MaxGroenenboom) [inline catkin_ws/src/kpr_interface/srv/SettingsIO.srv]: Typo

File: catkin_ws/src/kpr_interface/srv/SettingsIO.srv [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+uint8 succes
```
Comment (MaxGroenenboom) [inline catkin_ws/src/kpr_interface/srv/SettingsIO.srv]: Another typo :P

**Review by MaxGroenenboom — CHANGES_REQUESTED** [review]
I haven't tested the code yet, but it looks good.

File: catkin_ws/src/kpr_interface/cfg/settings.json (hunk 1 above)
Comment (ChielBruin) [reply catkin_ws/src/kpr_interface/cfg/settings.json:1]: They are strings to be able to support multiple data formats and primarily to simplify the communication.

File: catkin_ws/src/kpr_interface/srv/SettingsIO.srv (hunk 2 above)
Comment (ChielBruin) [reply catkin_ws/src/kpr_interface/srv/SettingsIO.srv]: Oops!

**Comment by MaxGroenenboom** [comment]
Why not load the settings in the interface and send it to the settings manager instead? The interface already supports editing the settings.

**Comment by ChielBruin** [comment]
The idea of the settings manager is that it performs all operations on the settings and provides an abstraction layer for all nodes that want to use this functionality. Therefore the settings manager is the correct location for the saving and loading of the settings.

**Comment by MaxGroenenboom** [comment]
Works as it should :+1:

**Review by MaxGroenenboom — APPROVED** [review]

**Pull request merged**